internal/test/wsu/wsu_test.go

```diff
@@ -55,75 +56,228 @@ var (
+
+	// Run the playbook without the pinned version option
+	cmd = exec.Command("ansible-playbook", "-v", playbookPath)
+	out, err = cmd.CombinedOutput()
+	require.NoError(t, err, "WSU playbook returned error: %s, with output: %s", err, string(out))
+
+	// Test that we built the wmcb
+	require.Contains(t, string(out),
+		"CGO_ENABLED=0 GO111MODULE=on GOOS=windows go build -o wmcb.exe  "+
+			"github.com/openshift/windows-machine-config-operator/cmd/bootstrapper")
+}
+
 // TestWSU creates a Windows instance, runs the WSU, and then runs a series of tests to ensure all expected
 // behavior was achieved. The following environment variables must be set for this test to run: KUBECONFIG,
 // AWS_SHARED_CREDENTIALS_FILE, ARTIFACT_DIR, KUBE_SSH_KEY_PATH, WSU_PATH, CLUSTER_ADDR
 func TestWSU(t *testing.T) {
 	require.NotEmptyf(t, playbookPath, "WSU_PATH environment variable not set")
 	require.NotEmptyf(t, clusterAddress, "CLUSTER_ADDR environment variable not set")
 
-	for _, vm := range framework.WinVMs {
-		runWSUTestSuite(t, vm)
-	}
-}
+	require.GreaterOrEqual(t, vmCount, 1, "Expected one or more VMs")
 
-// runWSUTestSuite runs the WSU test suite against the VM
-func runWSUTestSuite(t *testing.T, vm e2ef.WindowsVM) {
 	// In order to run the ansible playbook we create an inventory file:
 	// https://docs.ansible.com/ansible/latest/user_guide/intro_inventory.html
-	require.NotNil(t, vm.GetCredentials())
-	hostFilePath, err := createHostFile(vm.GetCredentials().GetIPAddress(),
-		vm.GetCredentials().GetPassword())
+	hostFilePath, err := createHostFile()
 	require.NoErrorf(t, err, "Could not write to host file: %s", err)
-	cmd := exec.Command("ansible-playbook", "-vvv", "-i", hostFilePath, playbookPath)
+
+	// Run the playbook
+	cmd := exec.Command("ansible-playbook", "-v", "-i", hostFilePath, playbookPath)
 	out, err := cmd.CombinedOutput()
 	require.NoError(t, err, "WSU playbook returned error: %s, with output: %s", err, string(out))
 
-	// Ansible will copy files to a temporary directory with a path such as:
-	// C:\\Users\\Administrator\\AppData\\Local\\Temp\\ansible.z5wa1pc5.vhn\\
-	initialSplit := strings.Split(string(out), "C:\\\\Users\\\\Administrator\\\\AppData\\\\Local\\\\Temp\\\\ansible.")
-	require.True(t, len(initialSplit) > 1, "Could not find Windows temp dir: %s", out)
-	ansibleTempDir = "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ansible." + strings.Split(initialSplit[1], "\"")[0]
+	// Run cluster wide tests
+	t.Run("Pending CSRs were approved", testNoPendingCSRs)
+
+	// Run VM specific tests
+	for i, vm := range framework.WinVMs {
+		func(vm e2ef.WindowsVM) {
+			t.Run("VM "+strconv.Itoa(i), func(t *testing.T) {
+				// Indicate that we can run the test suite on each node in parallel
+				t.Parallel()
+				runVMTestSuite(t, vm)
+			})
+		}(vm)
+	}
+}
+
+// getAnsibleTempDirPath returns the path of the ansible temp directory on the remote VM
+func getAnsibleTempDirPath(vm e2ef.WindowsVM) (string, error) {
```

Comment:
This seems fragile. It might be safer to get the temp directory from the ansible output.